### PR TITLE
ZTS: Misc test fixes for FreeBSD

### PR DESCRIPTION
**Pull request by freqlabs**
### Motivation and Context

Several test scripts follow a pattern `if is_linux; then linux things; else illumos things; fi` where the illumos things do not make sense for FreeBSD.

### Description

Add missing logic for FreeBSD to a few test scripts.

### How Has This Been Tested?

These changes have been running through our CI on FreeBSD.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [x] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [x] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [x] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

File: tests/zfs-tests/tests/functional/pool_checkpoint/checkpoint_discard_busy.ksh [hunk 1]
```diff
@@ -15,6 +15,7 @@
 # Copyright (c) 2017, 2018 by Delphix. All rights reserved.
 #
 
+. $STF_SUITE/include/libtest.shlib
```
Comment (jwk404) [inline tests/zfs-tests/tests/functional/pool_checkpoint/checkpoint_discard_busy.ksh]: What is this for?

File: tests/zfs-tests/tests/functional/rsend/rsend_012_pos.ksh [hunk 2]
```diff
@@ -26,7 +26,7 @@
 # Use is subject to license terms.
 #
 
-#
+. $STF_SUITE/include/libtest.shlib
```
Comment (jwk404) [inline tests/zfs-tests/tests/functional/rsend/rsend_012_pos.ksh]: What is this for?

File: tests/zfs-tests/tests/functional/pool_checkpoint/checkpoint_capacity.ksh [hunk 3]
```diff
@@ -15,6 +15,7 @@
 # Copyright (c) 2017 by Delphix. All rights reserved.
 #
 
+. $STF_SUITE/include/libtest.shlib
```
Comment (jwk404) [inline tests/zfs-tests/tests/functional/pool_checkpoint/checkpoint_capacity.ksh]: What is this for?

File: tests/zfs-tests/tests/functional/trim/setup.ksh [hunk 4]
```diff
@@ -24,6 +24,7 @@
 verify_runnable "global"
 
 if is_freebsd; then
+	log_unsupported "FreeBSD has no hole punching mechanism for the time being."
 	diskinfo -v $DISKS | grep -qE 'No.*# TRIM/UNMAP support' &&
```
Comment (jwk404) [inline tests/zfs-tests/tests/functional/trim/setup.ksh]: The diskinfo command is dead code now.

Comment (freqlabs) [reply tests/zfs-tests/tests/functional/trim/setup.ksh]: I left it for when we do enable these tests.

File: tests/zfs-tests/tests/functional/pool_checkpoint/checkpoint_discard_busy.ksh (hunk 1 above)
Comment (freqlabs) [reply tests/zfs-tests/tests/functional/pool_checkpoint/checkpoint_discard_busy.ksh]: Dropped

File: tests/zfs-tests/tests/functional/rsend/rsend_012_pos.ksh (hunk 2 above)
Comment (freqlabs) [reply tests/zfs-tests/tests/functional/rsend/rsend_012_pos.ksh]: Dropped

File: tests/zfs-tests/tests/functional/pool_checkpoint/checkpoint_capacity.ksh (hunk 3 above)
Comment (freqlabs) [reply tests/zfs-tests/tests/functional/pool_checkpoint/checkpoint_capacity.ksh]: Dropped

**Pull request merged**